### PR TITLE
Adaptive UI: added a recipe helper for interactive color sets

### PR DESCRIPTION
**Pull request by bheston**
# Pull Request

## 📖 Description

I'd like some feedback on this. I'm adjusting the token structure a bit and am looking for a way to make it less error-prone and less reliant on copy and paste. This new helper method sets up a common relationship between a `recipe`, `set` (new), and css tokens for `rest`, `hover`, `active`, and `focus`.

This will pave the way for introducing `set` and eventually migrating a `disabled` color into the `set` and tokens.

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: packages/utilities/adaptive-ui/src/design-tokens/color.ts [hunk 1]
```diff
@@ -63,11 +64,8 @@ export const accentFillFocusDelta = createNonCss<number>(
     "accent-fill-focus-delta"
 ).withDefault(0);
 
-/** @public */
-export const accentFillRecipe = createNonCss<InteractiveColorRecipe>(
-    "accent-fill-recipe"
-).withDefault({
-    evaluate: (element: HTMLElement, reference?: Swatch): InteractiveSwatchSet =>
+const accentFill = createInteractiveColorTokens(
```
Comment (KingOfTac) [inline packages/utilities/adaptive-ui/src/design-tokens/color.ts:67]: How does having all of this in a set affect tree shaking? Are the cases where someone would want to import only `accentFillRest` and expect the other accent interactive tokens to be tree shaken away, or is the assumption that if you import one, then you'll need to import the rest in order to have a consistent look across element states?

Comment (bheston) [reply packages/utilities/adaptive-ui/src/design-tokens/color.ts:67]: My thought toward addressing this is that all instances of creating those individual tokens has been moved into a single instance of the function above. Yes, this means that if you use _any_ `interactiveColorTokens` you get the code for `rest`, `hover`, etc. That code above could probably be consolidated as well.
At least if you use even _two_ `interactiveColorTokens` you get to share the code to create those stateful tokens.
I think that actually would make code size smaller, but I haven't tested it.

Comment (bheston) [reply packages/utilities/adaptive-ui/src/design-tokens/color.ts:67]: Actually, `rest` etc. above could even be functions that cache the output. Then if you never use a `hover` state or whatever, you never `create` the `DesignToken`. That seems efficient from the perspective of the amount of code running compared to what is used.

File: packages/utilities/adaptive-ui/src/color/recipe.ts [hunk 2]
```diff
@@ -56,3 +58,58 @@ export interface InteractiveSwatchSet {
...
+
+        set: setToken,
+
+        rest: create<Swatch>(`${name}-rest`).withDefault(
+            (element: HTMLElement) => setToken.getValueFor(element).rest
+        ),
+
+        hover: create<Swatch>(`${name}-hover`).withDefault(
+            (element: HTMLElement) => setToken.getValueFor(element).hover
+        ),
```
Comment (bheston) [inline packages/utilities/adaptive-ui/src/color/recipe.ts:105]: Per below, would this property be better as a cached function, where the design tokens is not created until the first time it's referenced?

Comment (nicholasrice) [reply packages/utilities/adaptive-ui/src/color/recipe.ts:105]: Yeah, assigning a default value will effectively prevent any tree-shaking which may be undesirable.

**Pull request closed without merging**